telosnetwork/teloscan
# Issue 141 | Testing setup

# Addresses #141 

## Notes
This PR adds in Jest (and the necessary configuration changes) to enable unit testing, as well as a simple snapshot test for verification purposes

## Testing Notes
- from the root of the project, open a console
- run `yarn install` to add new dependencies
- run `yarn test`
    - you should see Jest output information saying a single test has passed
- make any edit to the `<template>` of `src/components/Footer.vue`, e.g. changing _Teloscan_ to _Teloscan't_
- re-run `yarn test`
    - you should see one failed test in your console informing you of precisely why it failed (in this case, the markup will not match the snapshot, and you should see the difference in markup printed out)
- run `git reset --hard` to revert your changes
- run `git checkout -b fake-test-branch && git commit --allow-empty -m "Empty-Commit" && git push -u origin fake-test-branch` to create a branch with an empty commit and push it to origin
    - you should see a successful test run in the console
- run `git checkout testing-setup && git branch -d fake-test-branch && git push origin --delete fake-test-branch` to checkout the PR branch and delete the mock branch locally & remotely
